src/bootstrap/config.rs

```diff
@@ -798,7 +798,20 @@ impl Config {
         config.llvm_profile_use = flags.llvm_profile_use;
         config.llvm_profile_generate = flags.llvm_profile_generate;
 
-        let stage0_json = t!(std::fs::read(&config.src.join("src").join("stage0.json")));
+        let stage0_json = std::fs::read(&config.src.join("src").join("stage0.json"))
+            .unwrap_or_else(|_| {
+                let base_path = output(Command::new("git").arg("rev-parse").arg("--show-toplevel"))
+                    .trim()
+                    .to_owned();
+
+                config.src = PathBuf::from(base_path);
+                config.initial_rustc =
+                    config.src.join("build").join(config.build.triple).join("stage0/bin/rustc");
+                config.initial_cargo =
+                    config.src.join("build").join(config.build.triple).join("stage0/bin/cargo");
```

Comment:
this should be `config.out` instead, the build directory isn't always within the source checkout (particularly in CI).
```suggestion
                    config.out.join(config.build.triple).join("stage0/bin/rustc");
                config.initial_cargo =
                    config.out.join(config.build.triple).join("stage0/bin/cargo");
```